src/scenarios/shared/mauisharedpython.py

```diff
@@ -139,6 +159,145 @@ def extract_latest_dotnet_feed_from_nuget_config(path: str, offset: int = 0) ->
 
     return target_feed
 
+def download_maui_nuget_config(target_framework: str = "net10.0", output_filename: str = "MauiNuGet.config") -> str:
+    '''
+        Download MAUI's NuGet.config from the appropriate branch.
+        Returns the path to the downloaded config file.
+        
+        Args:
+            target_framework: Target framework to determine which branch to use (e.g., "net10.0")
+            output_filename: Name of the file to save the downloaded config
+    '''
+    # Extract base framework version (e.g., "net10.0" from "net10.0-android")
+    if '-' in target_framework:
+        target_framework_wo_platform = target_framework.split('-')[0]
+    else:
+        target_framework_wo_platform = target_framework
+    
+    url = f'https://raw.githubusercontent.com/dotnet/maui/{target_framework_wo_platform}/NuGet.config'
+    getLogger().info(f"Downloading MAUI NuGet.config from {url}")
```

Comment:
@PureWeen @rmarinho is the branch name on dotnet/maui changing?

This maps `$(TargetFramework)` == branch. Does it matter if they use dotnet/maui/main vs net10.0?